flagd/pkg/service/flag-sync/sync-multiplexer.go

```diff
@@ -0,0 +1,193 @@
+package sync
+
+import (
+	"encoding/json"
+	"fmt"
+	"slices"
+	"strings"
+	"sync"
+
+	"github.com/open-feature/flagd/core/pkg/model"
+	"github.com/open-feature/flagd/core/pkg/store"
+)
+
+// Multiplexer abstract subscription handling and storage processing.
+// Flag configurations will be lazy loaded using reFill logic upon the calls to publish.
+type Multiplexer struct {
+	store   *store.Flags
+	sources []string
+
+	subs         map[interface{}]subscription            // subscriptions on all sources
+	selectorSubs map[string]map[interface{}]subscription // source specific subscriptions
+
+	allFlags      string            // pre-calculated all flags in store as a string
+	selectorFlags map[string]string // pre-calculated selector scoped flags in store as strings
+
+	mu sync.Mutex
+}
+
+type subscription struct {
+	id      interface{}
+	channel chan payload
+}
+
+type payload struct {
+	flags string
+}
+
+// NewMux creates a new sync multiplexer
+func NewMux(store *store.Flags, sources []string) (*Multiplexer, error) {
+	m := &Multiplexer{
+		store:         store,
+		sources:       sources,
+		subs:          map[interface{}]subscription{},
+		selectorSubs:  map[string]map[interface{}]subscription{},
+		selectorFlags: map[string]string{},
+	}
+
+	return m, m.reFill()
+}
+
+// Register a subscription
+func (r *Multiplexer) Register(id interface{}, source string, con chan payload) error {
+	r.mu.Lock()
+	defer r.mu.Unlock()
+
+	if source != "" && !slices.Contains(r.sources, source) {
+		return fmt.Errorf("no flag watcher setup for source %s", source)
+	}
+
+	var initSync string
+
+	if source == "" {
```

Comment:
do we document this behavior? I was under the impression that if we don't provide any source then we don't return any flags

Reply:
This behaviour is agreed and defined through the issue itself :)